### PR TITLE
feature/milestone1-spring, add spring-security, add API Documentation

### PR DESCRIPTION
**Pull request by Jeffchiucp**
- add API Documentation (swagger UI)
-add cors filter.
-exception handling.
-add spring-security.

### CODE REVIEW

**Comment by Jeffchiucp** [comment]
SwaggerAPIDocumentation
You can run it locally at localhost 
at the following URL: http://localhost:8080

**Comment by Jeffchiucp** [comment]
You will need to install maven 3.x version then check the snapshot.

**Comment by Jeffchiucp** [comment]
You can use this link http://localhost:8080/swagger-ui-html to check out the API Documentation. 
Most of the documentation will be based on https://raw.githubusercontent.com/arrangement-io/arrangespace-server/master/doc/api_data.js

**Comment by Jeffchiucp** [comment]
You run the Spring app using the following command:
mvn spring-boot:run

**Comment by natyconnor** [comment]
You seem to have inconsistent tabbing. Java always uses real tabs

**Review by deonjo — APPROVED** [review]

**Comment by Jeffchiucp** [comment]
@natyconnor where do you see inconsistent tabbing?

File: src/main/java/com/arrangespace/dao/UserDao.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.arrangespace.model.User;
+
+@Repository
+public interface UserDao extends MongoRepository<User, String> {
+	
+	public Optional<User> findByGoogleId(String googleId);
+	
+	public Optional<User> findByEmail(String email);
+
+
```
Comment (natyconnor) [inline src/main/java/com/arrangespace/dao/UserDao.java]: extra new line

**Review by natyconnor — COMMENTED** [review]
I see spaces in
`SecurityConfig.java` (lines 18-20)
`WebMvcConfig.java` (all of it?)
`BadRequestException`
`ResourceNotFoundException`
`CookieUtils`

**Comment by Jeffchiucp** [comment]
got it. Let me fixed that.

**Comment by Jeffchiucp** [comment]
This is my editor setting in eclipse. Any suggestion?
image

**Comment by Jeffchiucp** [comment]
@natyconnor Do u have any others policy? suggestion? eg: https://github.com/google/google-java-format

**Pull request merged**